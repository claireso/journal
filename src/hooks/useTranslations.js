import React from 'react'
import PropTypes from 'prop-types'

import translations from '../translations.json'

const TranslationsContext = React.createContext({})

export const TranslationsProvider = ({ children, translations }) => (
  <TranslationsContext.Provider value={translations}>{children}</TranslationsContext.Provider>
)

TranslationsProvider.propTypes = {
  children: PropTypes.node,
  translations: PropTypes.object
}

export const useTranslations = () => React.useContext(TranslationsContext)

export const getTranslations = (lang, namespace) => {
  const dict = translations[lang]

  if (!dict) {
    throw new Error(`Language '${lang}' is not supported by the application`)
  }

  if (!namespace) return dict

  const trans = dict[namespace]

  if (!trans) {
    throw new Error(`Namespace '${namespace}' is not defined for the language ${lang}`)
  }

  return trans
}
