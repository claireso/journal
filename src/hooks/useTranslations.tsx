'use client'

import React from 'react'
import PropTypes from 'prop-types'

import { getTranslations } from '../translations'

const TranslationsContext = React.createContext({})

type Translations = {
  [key: string]: object | string
}

interface TranslationsProviderProps {
  children: React.ReactNode
  namespace: 'client' | 'admin'
}

export const TranslationsProvider = ({ children, namespace }: TranslationsProviderProps) => {
  const translations = getTranslations(process.env.NEXT_PUBLIC_WEBSITE_LANGUAGE, namespace)

  return <TranslationsContext.Provider value={translations}>{children}</TranslationsContext.Provider>
}

TranslationsProvider.propTypes = {
  children: PropTypes.node,
  translations: PropTypes.object
}

export const useTranslations = (): Translations => React.useContext(TranslationsContext)
