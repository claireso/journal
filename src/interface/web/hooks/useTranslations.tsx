'use client'

import React from 'react'
import PropTypes from 'prop-types'

import { getTranslations } from '@infrastructure/translations'

const TranslationsContext = React.createContext({})

type Translations = {
  [key: string]: object | string
}

interface TranslationsProviderProps {
  children: React.ReactNode
  namespace: 'client' | 'admin'
  lang?: string
}

export const TranslationsProvider = ({ children, namespace, lang }: TranslationsProviderProps) => {
  const translations = getTranslations(lang ?? 'en', namespace)

  return <TranslationsContext.Provider value={translations}>{children}</TranslationsContext.Provider>
}

TranslationsProvider.propTypes = {
  children: PropTypes.node,
  translations: PropTypes.object
}

export const useTranslations = (): Translations => React.useContext(TranslationsContext)
