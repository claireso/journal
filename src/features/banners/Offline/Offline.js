import React from 'react'

import { useTranslations } from '@hooks/useTranslations'
import useIsOnline from '@hooks/useIsOnline'

import Flash from '@components/Flash'

const BannerOffline = () => {
  const isOnline = useIsOnline()
  const translations = useTranslations()

  if (isOnline) return null

  return (
    <Flash status="info" css={{ mb: 0 }}>
      {translations.bannerOffline}
    </Flash>
  )
}

export default BannerOffline
