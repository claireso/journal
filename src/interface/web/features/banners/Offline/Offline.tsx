'use client'

import React from 'react'

import { useTranslations } from '@web/hooks/useTranslations'
import useIsOnline from '@web/hooks/useIsOnline'

import Flash from '@web/components/Flash'

const BannerOffline = () => {
  const isOnline = useIsOnline()
  const translations = useTranslations()

  if (isOnline) return null

  return <Flash status="info">{translations.bannerOffline as string}</Flash>
}

export default BannerOffline
