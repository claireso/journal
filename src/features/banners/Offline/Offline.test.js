import { render } from '@testing-library/react'

import { TranslationsProvider, getTranslations } from '@hooks/useTranslations'
import Offline from './index'

describe('<Offline />', () => {
  const renderBanner = () =>
    render(
      <TranslationsProvider translations={getTranslations('en', 'client')}>
        <Offline />
      </TranslationsProvider>
    )

  test('should not render component', () => {
    const { container } = renderBanner()

    expect(container).toMatchSnapshot()
  })

  test('should render component', () => {
    global.goOffline()

    const { container } = renderBanner()

    expect(container).toMatchSnapshot()

    global.goOnline()
  })
})
