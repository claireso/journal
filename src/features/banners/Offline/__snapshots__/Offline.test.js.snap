// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Offline /> should not render component 1`] = `<DocumentFragment />`;

exports[`<Offline /> should render component 1`] = `
<DocumentFragment>
  <div
    class="c-fyLyTf c-fyLyTf-kHVEuQ-status-info c-fyLyTf-ibVwYlk-css"
  >
    You are currently offline. Connect to the internet for access to the latest photos
  </div>
</DocumentFragment>
`;
